### DISCUSSION TITLE
Dexcom G7 - Android 15 major connection issue

### DISCUSSION THREAD

**Discussion by ninelore** [NightscoutFoundation/xDrip#3735] [Bug reports] [answered]
Another thread as requested by @Navid200 

Similar to #3733, I only get a couple of readings and nothing afterwards until i reboot or re-pair my sensor (and sometimes not even after that)

[Dexcom acknowledged the issue](https://www.dexcom.com/en-US/faqs/will-android-15-affect-my-dexcom-g7-cgm-system), but the workaround described does not work, regardless of official app or xdrip

Some supposed dexcom dev stated on reddit that the issue is in Android 15 itself (w/o proof), but a renowned ROM dev suggested that a bad implementation might've been broken by security improvements made to the Android BT stack.

Logs ([with debug logs enabled](https://github.com/NightscoutFoundation/xDrip/discussions/3733#discussioncomment-11059481)) have been sent via xdrip to jamorham.

xdrip 9f769df-2024.10.11
Pixel 9 Pro - tested both Stock Android 15 and GrapheneOS 15 2024102400 / 2024102100

**Comment by Navid200** [comment]
Do you also have a pump or a Dexcom receiver or a BlueJay watch?

**Reply by ninelore** [reply]
Omnipod Dash via AAPS (works fine)

**Reply by Navid200** [reply]
But, that pump does not communicate with G7, right?

**Reply by ninelore** [reply]
nope

**Reply by Navid200** [reply]
Is your battery low also?  Can you post a screenshot of the system status page?

**Reply by ninelore** [reply]
[Screenshot_20241026-140207.png](https://github.com/user-attachments/assets/4c5acf5b-90f4-4653-a144-f7040263312c)

**Reply by Navid200** [reply]
Have you restarted the phone with this sensor?
I have to restart mine (Pixel 6a) with every new sensor.

**Reply by ninelore** [reply]
Yes, over a dozen times

**Reply by ninelore** [reply]
tbf it could be that stuff breaks as soon as something else connects via BT

**Reply by ninelore** [reply]
It did indeed work for a while after i disabled aaps. Its stopped working as soon as i connected BT headphones, confirming my suspicion

**Comment by ninelore** [comment]
Some further infodump

- The issue first occured in Beta 4 on Android 15

- Same Issue on Pixel 6 Pro, both Stock 15 and well as GrapheneOS and XDrip as well as the official App

**Comment by Navid200** [comment]
Does Omnipod dash get glucose readings?  If yes, how?

**Reply by ninelore** [reply]
Nope, Dash is just a simple pump controlled via BT from official handheld or aaps

**Reply by Navid200** [reply]
So, xDrip gets readings from G7, sends them to AAPS, and AAPS uses Bluetooth to control the pump?

**Reply by ninelore** [reply]
Yes, however not having AAPS installed still yields the same result

**Comment by ninelore** [comment, marked as answer]
cannot reproduce anymore
